postgres-protocol/src/replication/mod.rs

```diff
@@ -0,0 +1,33 @@
+//! This module defines the [DecodingPlugin] trait that can be used to implement the decoding
+//! logic of any logical decoding plugin.
+//!
+//! Two reference implementations are provided:
+//!
+//! * [PgOutput]: Decodes the built-in pgoutput stream
+//! * [Raw]: A dummy plugin that yield each message it receives as bytes
```

Comment:
This seems like a much higher level API than what's intended to be in the postgres-protocol crate.

Reply:
We can move in different directions here. The first of the implementations, `pgoutput`, is part of the postgres protocol (https://www.postgresql.org/docs/10/protocol-logicalrep-message-formats.html) so I think you would agree that at least that one belongs to this crate.

Now, initially I had it set up such that this was the only mode of logical replication that you could use, but in fact there are many plugins out there, so I added this trait.

What do you think of removing this trait completely and just offering two concrete ways of getting a replication stream? One that gives you the decoded built-in messages of postgres, and one that gives you bytes. And anything that needs to parse the bytes can do so externally without messing with traits of this crate.